internal/v4/stats.go

```diff
@@ -132,17 +132,35 @@ func (h *ReqHandler) serveStatsUpdate(_ http.Header, r *http.Request) (interface
 		return nil, errgo.Notef(err, "cannot unmarshal body")
 	}
 
-	rid, err := h.resolveURL(req.CharmReference)
-	if err != nil {
-		return nil, errgo.Mask(err, errgo.Is(params.ErrNotFound))
-	}
+	errors := make([]error, 0)
+	for i := range req {
+		rid, err := h.resolveURL(req[i].CharmReference)
+		if err != nil {
+			errors = append(errors, errgo.Notef(err, "cannot find entity for url: %s", req[i].CharmReference))
```

Comment:
Perhaps:
    errors = append(errors, errgo.Notef(err, "cannot find entity for %s", req[i].CharmReference))

?
(we usually use the : for underlying errors)